cscs-checks/prgenv/affinity_check.py

```diff
@@ -0,0 +1,155 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class AffinityTestBase(rfm.RegressionTest):
+    def __init__(self, variant):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system = 'Make'
+        self.build_system.options = ['--directory=affinity', 'MPI=1']
+        self.prebuild_cmd = ['git clone https://github.com/vkarak/affinity']
+        self.executable = 'affinity/affinity'
+        self.variant = variant
+
+        self.maintainers = ['RS', 'VK']
+        self.tags = {'production', 'scs'}
+
+    def setup(self, partition, environ, **job_opts):
+
+        def parse_affinity(x):
+            x_int = [int(xi) for xi in x.split()]
+            return sorted(x_int)
+
+        re_aff_cores = r'CPU affinity: \[\s+(?P<cpu>[\d+\s+]+)\]'
+        self.aff_cores = sn.extractall(
+            re_aff_cores, self.stdout, 'cpu', parse_affinity)
+        self.ref_cores = sn.extractall(
+            re_aff_cores, self.cases[self.variant]['ref_%s' % partition.name],
+            'cpu', parse_affinity)
+        re_aff_thrds = r'^Tag:[^\n\r]*Thread:\s+(?P<thread>\d+)'
+        self.aff_thrds = sn.extractall(re_aff_thrds, self.stdout, 'thread',
+                                       int)
+        self.ref_thrds = sn.extractall(
```

Comment:
I know that you wanted to save some space and be more compact, but I could not understand the name of the variable from the beginning. I would vote for ref_num_threads as variable name. Am I being too picky?

The same applies to the other cases.

Reply:
@victorusu I don't prefer very long names.